rust/main/chains/hyperlane-ethereum/src/rpc_clients/provider.rs

```diff
@@ -83,6 +90,23 @@ where
     ) -> ChainResult<u32> {
         get_finalized_block_number(&*self.provider, reorg_period).await
     }
+
+    async fn send(&self, tx: TypedTransaction, function: Function) -> ChainResult<H256> {
```

Comment:
nit: could we pass `EthereumTxPrecursor` instead of `tx` and `function`?

Reply:
We'll need to move `EthereumTxPrecursor` from Lander to `hyperlane-ethereum`. It is possible, just creates a dependency beyond RPC calls.

Reply:
ah makes sense